<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (19) on Tue Aug 22 11:59:40 BST 2023 -->
<title>C-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2023-08-22">
<meta name="description" content="index: C">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.6.0.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:C">C</h2>
<dl class="index">
<dt><a href="../Algorithms/CalculateOverlaps.html" class="type-name-link" title="class in Algorithms">CalculateOverlaps</a> - Class in <a href="../Algorithms/package-summary.html">Algorithms</a></dt>
<dd>
<div class="block">Calculates and saved query contigs which overlap in regions of their alignment.</div>
</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#%3Cinit%3E()" class="member-name-link">CalculateOverlaps()</a> - Constructor for class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#calculateRefOverlap(java.lang.String,DataTypes.RefContig,java.util.LinkedHashMap)" class="member-name-link">calculateRefOverlap(String, RefContig, LinkedHashMap&lt;String, QryContig&gt;)</a> - Static method in class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>
<div class="block">Calculates overlapping regions</div>
</dd>
<dt><a href="../Algorithms/DetectSV.html#changeOrientation(java.util.Map)" class="member-name-link">changeOrientation(Map&lt;Integer, List&lt;Integer&gt;&gt;)</a> - Static method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Cigar.html#changeOrientation(java.util.Map)" class="member-name-link">changeOrientation(Map&lt;Integer, List&lt;Integer&gt;&gt;)</a> - Static method in class DataTypes.<a href="../DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></dt>
<dd>
<div class="block">Change orientation of query alignments</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.html#checkEnzymeName(javax.swing.JTextField)" class="member-name-link">checkEnzymeName(JTextField)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.html" title="class in ComparativeGenomics.Panels">CompGenStart</a></dt>
<dd>
<div class="block">Checks if enzyme name from enzyme text field is empty</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.html#checkEnzymeSite(javax.swing.JTextField)" class="member-name-link">checkEnzymeSite(JTextField)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.html" title="class in ComparativeGenomics.Panels">CompGenStart</a></dt>
<dd>
<div class="block">Checks if enzyme site is empty.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.html#checkSite(java.lang.String)" class="member-name-link">checkSite(String)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.html" title="class in ComparativeGenomics.Panels">CompGenStart</a></dt>
<dd>
<div class="block">Check if a digestion site is valid, ie only contains nucleotides (A, T,
 C, G)</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" class="type-name-link" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a> - Class in <a href="../ComparativeGenomics/FileHandling/DataHandling/package-summary.html">ComparativeGenomics.FileHandling.DataHandling</a></dt>
<dd>
<div class="block">Combines the CmapData, XmapData, annotations (gene), structural variants and
 context of the chromosome within a genom</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#%3Cinit%3E(java.lang.String,ComparativeGenomics.FileHandling.DataHandling.CmapData,java.lang.Double,ComparativeGenomics.FileHandling.DataHandling.Sequence,java.util.ArrayList)" class="member-name-link">Chromosome(String, CmapData, Double, Sequence, ArrayList&lt;Gene&gt;)</a> - Constructor for class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Constructor of class Chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/ChromosomeChartPanel.html" class="type-name-link" title="class in ComparativeGenomics.Panels">ChromosomeChartPanel</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Plots the number of each different type of SV event per chromosome within the reference genome using XChart.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/ChromosomeChartPanel.html#%3Cinit%3E()" class="member-name-link">ChromosomeChartPanel()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/ChromosomeChartPanel.html" title="class in ComparativeGenomics.Panels">ChromosomeChartPanel</a></dt>
<dd>
<div class="block">Constructor</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/ChromosomePanel.html" class="type-name-link" title="class in ComparativeGenomics.Panels">ChromosomePanel</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Extends javax.swing.JPanel which draws all of the XmapData
 objects corresponding to a chosen Chromosome.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/ChromosomePanel.html#%3Cinit%3E()" class="member-name-link">ChromosomePanel()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/ChromosomePanel.html" title="class in ComparativeGenomics.Panels">ChromosomePanel</a></dt>
<dd>
<div class="block">Creates new ChromosomePanel</div>
</dd>
<dt><a href="../DataTypes/Cigar.html" class="type-name-link" title="class in DataTypes">Cigar</a> - Class in <a href="../DataTypes/package-summary.html">DataTypes</a></dt>
<dd>
<div class="block">Parse and save cigar content</div>
</dd>
<dt><a href="../DataTypes/Cigar.html#%3Cinit%3E(java.lang.String)" class="member-name-link">Cigar(String)</a> - Constructor for class DataTypes.<a href="../DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></dt>
<dd>
<div class="block">Class constructor with hitEnum</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CircosPanel.html" class="type-name-link" title="class in ComparativeGenomics.Panels">CircosPanel</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Draws the chromosome making up the genome in a circle.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CircosPanel.html#%3Cinit%3E()" class="member-name-link">CircosPanel()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CircosPanel.html" title="class in ComparativeGenomics.Panels">CircosPanel</a></dt>
<dd>
<div class="block">Creates new CircosPanel</div>
</dd>
<dt><a href="../Algorithms/DetectSV.html#classifyLabels()" class="member-name-link">classifyLabels()</a> - Method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Panels/QueryPanel.html#clearHighlightedGenes()" class="member-name-link">clearHighlightedGenes()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></dt>
<dd>
<div class="block">Clears all red borders of <code>Gene</code> objects in this panel</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/ChromosomePanel.html#clearSelection()" class="member-name-link">clearSelection()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/ChromosomePanel.html" title="class in ComparativeGenomics.Panels">ChromosomePanel</a></dt>
<dd>
<div class="block">Clears selected alignments</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html" class="type-name-link" title="class in ComparativeGenomics.FileHandling">Cmap</a> - Class in <a href="../ComparativeGenomics/FileHandling/package-summary.html">ComparativeGenomics.FileHandling</a></dt>
<dd>
<div class="block">Stores all of the contiguous maps contained within the
 cmap file the data are stored into hashmaps to allow for efficient access.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#%3Cinit%3E(java.lang.String)" class="member-name-link">Cmap(String)</a> - Constructor for class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Reads and parses cmap file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" class="type-name-link" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a> - Class in <a href="../ComparativeGenomics/FileHandling/DataHandling/package-summary.html">ComparativeGenomics.FileHandling.DataHandling</a></dt>
<dd>
<div class="block">Stores the data corresponding to each cmap ID within a CMAP file.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html#%3Cinit%3E(java.lang.Integer,java.lang.Integer,java.lang.Double)" class="member-name-link">CmapData(Integer, Integer, Double)</a> - Constructor for class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></dt>
<dd>
<div class="block">Constructor with ID, number of sites and length of cmap maps</div>
</dd>
<dt><a href="../FileHandling/CmapReader.html" class="type-name-link" title="class in FileHandling">CmapReader</a> - Class in <a href="../FileHandling/package-summary.html">FileHandling</a></dt>
<dd>
<div class="block">Validates and reads CMAP files</div>
</dd>
<dt><a href="../FileHandling/CmapReader.html#%3Cinit%3E()" class="member-name-link">CmapReader()</a> - Constructor for class FileHandling.<a href="../FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/CmapReader.html#cmapToHashMap(java.lang.String)" class="member-name-link">cmapToHashMap(String)</a> - Static method in class FileHandling.<a href="../FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></dt>
<dd>
<div class="block">Reads CMAP input file and parses it into a <code>HashMap</code> object</div>
</dd>
<dt><a href="../Algorithms/Variants.html#colorCigSites(java.util.Map,java.util.Set,java.lang.Double,java.lang.Double)" class="member-name-link">colorCigSites(Map&lt;Integer, Double&gt;, Set&lt;Integer&gt;, Double, Double)</a> - Method in class Algorithms.<a href="../Algorithms/Variants.html" title="class in Algorithms">Variants</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Drawing/package-summary.html">ComparativeGenomics.Drawing</a> - package ComparativeGenomics.Drawing</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/package-summary.html">ComparativeGenomics.FileHandling</a> - package ComparativeGenomics.FileHandling</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/package-summary.html">ComparativeGenomics.FileHandling.DataHandling</a> - package ComparativeGenomics.FileHandling.DataHandling</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a> - package ComparativeGenomics.Panels</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/package-summary.html">ComparativeGenomics.ServerHandling</a> - package ComparativeGenomics.ServerHandling</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/package-summary.html">ComparativeGenomics.StructuralVariant</a> - package ComparativeGenomics.StructuralVariant</dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.html" class="type-name-link" title="class in ComparativeGenomics.Panels">CompGenStart</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Provides the GUI and methods to start an alignment Job on an ExternalServer.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.html#%3Cinit%3E()" class="member-name-link">CompGenStart()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.html" title="class in ComparativeGenomics.Panels">CompGenStart</a></dt>
<dd>
<div class="block">Creates new form comparativeGenomics</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" class="type-name-link" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Model of table of jobs</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenView.html" class="type-name-link" title="class in ComparativeGenomics.Panels">CompGenView</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Panel on which alignments of two genomes using optical mapping data are displayed.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenView.html#%3Cinit%3E()" class="member-name-link">CompGenView()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenView.html" title="class in ComparativeGenomics.Panels">CompGenView</a></dt>
<dd>
<div class="block">Creates new form GenomeView</div>
</dd>
<dt><a href="../FileHandling/FastaReader.html#concatenateStrings(java.util.ArrayList)" class="member-name-link">concatenateStrings(ArrayList&lt;String&gt;)</a> - Static method in class FileHandling.<a href="../FileHandling/FastaReader.html" title="class in FileHandling">FastaReader</a></dt>
<dd>
<div class="block">Concatenates array list of strings</div>
</dd>
<dt><a href="../DataTypes/SV.html#confidence" class="member-name-link">confidence</a> - Variable in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/ConflictFileReader.html" class="type-name-link" title="class in FileHandling">ConflictFileReader</a> - Class in <a href="../FileHandling/package-summary.html">FileHandling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/ConflictFileReader.html#%3Cinit%3E()" class="member-name-link">ConflictFileReader()</a> - Constructor for class FileHandling.<a href="../FileHandling/ConflictFileReader.html" title="class in FileHandling">ConflictFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/ConflictFileWriter.html" class="type-name-link" title="class in FileHandling">ConflictFileWriter</a> - Class in <a href="../FileHandling/package-summary.html">FileHandling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/ConflictFileWriter.html#%3Cinit%3E()" class="member-name-link">ConflictFileWriter()</a> - Constructor for class FileHandling.<a href="../FileHandling/ConflictFileWriter.html" title="class in FileHandling">ConflictFileWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/SSH.html#connectServer()" class="member-name-link">connectServer()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/SSH.html" title="class in ComparativeGenomics.ServerHandling">SSH</a></dt>
<dd>
<div class="block">Connects to the external server by opening an ssh session</div>
</dd>
<dt><a href="../DataTypes/ContigInfo.html" class="type-name-link" title="class in DataTypes">ContigInfo</a> - Class in <a href="../DataTypes/package-summary.html">DataTypes</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/ContigInfo.html#%3Cinit%3E(double,DataTypes.LabelInfo%5B%5D)" class="member-name-link">ContigInfo(double, LabelInfo[])</a> - Constructor for class DataTypes.<a href="../DataTypes/ContigInfo.html" title="class in DataTypes">ContigInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/QryContig.html#copy()" class="member-name-link">copy()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/RefContig.html#copy()" class="member-name-link">copy()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/QryContig.html#copy(double)" class="member-name-link">copy(double)</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ProgressMonitor.html#count(long)" class="member-name-link">count(long)</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ProgressMonitor.html" title="class in ComparativeGenomics.ServerHandling">ProgressMonitor</a></dt>
<dd>
<div class="block">Sets the value of the <code>JProgressBar</code> to equal the number of bytes of the
 file that has been transferred</div>
</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#countAllOverlaps(java.util.LinkedHashMap,java.util.LinkedHashMap)" class="member-name-link">countAllOverlaps(LinkedHashMap&lt;String, RefContig&gt;, LinkedHashMap&lt;String, QryContig&gt;)</a> - Static method in class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>
<div class="block">Count all overlapping regions</div>
</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#countAllOverlaps(java.util.List)" class="member-name-link">countAllOverlaps(List&lt;Double&gt;)</a> - Static method in class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>
<div class="block">Count all overlapping regions</div>
</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#countAllOverlaps(java.util.Map)" class="member-name-link">countAllOverlaps(Map&lt;Integer, List&lt;Double&gt;&gt;)</a> - Static method in class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>
<div class="block">Counts all overlapping regions class constructor</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/Curve2D.html" class="type-name-link" title="class in ComparativeGenomics.Drawing">Curve2D</a> - Class in <a href="../ComparativeGenomics/Drawing/package-summary.html">ComparativeGenomics.Drawing</a></dt>
<dd>
<div class="block">Draws curves between two chromosomes in circos view to graphically indicate a
 translocation event.</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/Curve2D.html#%3Cinit%3E(float,float,float,float,float,float,java.lang.String)" class="member-name-link">Curve2D(float, float, float, float, float, float, String)</a> - Constructor for class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/Curve2D.html" title="class in ComparativeGenomics.Drawing">Curve2D</a></dt>
<dd>
<div class="block">Constructor with curve size and positions</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
