<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (19) on Tue Aug 22 11:59:40 BST 2023 -->
<title>G-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2023-08-22">
<meta name="description" content="index: G">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.6.0.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:G">G</h2>
<dl class="index">
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" class="type-name-link" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a> - Class in <a href="../ComparativeGenomics/FileHandling/DataHandling/package-summary.html">ComparativeGenomics.FileHandling.DataHandling</a></dt>
<dd>
<div class="block">Stores the gene entries from a GTF or GFFF file via the <code>Annot</code>
 object.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#%3Cinit%3E()" class="member-name-link">Gene()</a> - Constructor for class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Constructor</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#%3Cinit%3E(java.lang.String,java.lang.String,java.lang.String,java.lang.Double,java.lang.Double,java.lang.String)" class="member-name-link">Gene(String, String, String, Double, Double, String)</a> - Constructor for class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Constructor with chromosome, type, source, start and end, information</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" class="type-name-link" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a> - Class in <a href="../ComparativeGenomics/FileHandling/DataHandling/package-summary.html">ComparativeGenomics.FileHandling.DataHandling</a></dt>
<dd>
<div class="block">Stores all the metadata for example species of the genome,
 amd group the files related to the genome.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#%3Cinit%3E(java.lang.String,ComparativeGenomics.FileHandling.Cmap,ComparativeGenomics.FileHandling.Karyotype,ComparativeGenomics.FileHandling.Fasta,ComparativeGenomics.FileHandling.Annot)" class="member-name-link">Genome(String, Cmap, Karyotype, Fasta, Annot)</a> - Constructor for class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Constructor with name, cmap, karyotype, fasta and annotation file information</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/GenomePanel.html" class="type-name-link" title="class in ComparativeGenomics.Panels">GenomePanel</a> - Class in <a href="../ComparativeGenomics/Panels/package-summary.html">ComparativeGenomics.Panels</a></dt>
<dd>
<div class="block">Displays all the chromosomes of the reference.</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/GenomePanel.html#%3Cinit%3E()" class="member-name-link">GenomePanel()</a> - Constructor for class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/GenomePanel.html" title="class in ComparativeGenomics.Panels">GenomePanel</a></dt>
<dd>
<div class="block">Creates new form GenomePanel</div>
</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getAlignment()" class="member-name-link">getAlignment()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets alignment</div>
</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getAlignmentInfo(java.lang.String)" class="member-name-link">getAlignmentInfo(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getAlignmentLen()" class="member-name-link">getAlignmentLen()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getAlignments()" class="member-name-link">getAlignments()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets all the alignments to this chromosome</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getAlignments()" class="member-name-link">getAlignments()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getAlignments()" class="member-name-link">getAlignments()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getAlignmentSites()" class="member-name-link">getAlignmentSites()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getAlignmentSites()" class="member-name-link">getAlignmentSites()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/SortOverlap.html#getAllQueryinfo(java.util.List)" class="member-name-link">getAllQueryinfo(List&lt;Query&gt;)</a> - Method in class Algorithms.<a href="../Algorithms/SortOverlap.html" title="class in Algorithms">SortOverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getAllXmaps()" class="member-name-link">getAllXmaps()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets all xmap from this file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getAnnot()" class="member-name-link">getAnnot()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets annotations for this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getAnnotations()" class="member-name-link">getAnnotations()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets all <code>Gene</code> objects added to this <code>Chromosome</code></div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/EnzymeChartPanel.html#getBestEnz()" class="member-name-link">getBestEnz()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/EnzymeChartPanel.html" title="class in ComparativeGenomics.Panels">EnzymeChartPanel</a></dt>
<dd>
<div class="block">Gets best enzyme</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/BestEnzyme.html#getBestEnzyme()" class="member-name-link">getBestEnzyme()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/BestEnzyme.html" title="class in ComparativeGenomics.FileHandling">BestEnzyme</a></dt>
<dd>
<div class="block">Return enzyme with highest density</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/SSH.html#getChannel()" class="member-name-link">getChannel()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/SSH.html" title="class in ComparativeGenomics.ServerHandling">SSH</a></dt>
<dd>
<div class="block">Gets the sftp channel.</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getChimQual()" class="member-name-link">getChimQual()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/LabelInfo.html#getChimQuality()" class="member-name-link">getChimQuality()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/QueryView.html#getChosenQry()" class="member-name-link">getChosenQry()</a> - Static method in class MapOptics.<a href="../MapOptics/QueryView.html" title="class in MapOptics">QueryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/ReferenceView.html#getChosenQry()" class="member-name-link">getChosenQry()</a> - Static method in class MapOptics.<a href="../MapOptics/ReferenceView.html" title="class in MapOptics">ReferenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/SVView.html#getChosenQry()" class="member-name-link">getChosenQry()</a> - Static method in class MapOptics.<a href="../MapOptics/SVView.html" title="class in MapOptics">SVView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/DetectSV.html#getChosenRef()" class="member-name-link">getChosenRef()</a> - Static method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/QueryView.html#getChosenRef()" class="member-name-link">getChosenRef()</a> - Static method in class MapOptics.<a href="../MapOptics/QueryView.html" title="class in MapOptics">QueryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/ReferenceView.html#getChosenRef()" class="member-name-link">getChosenRef()</a> - Static method in class MapOptics.<a href="../MapOptics/ReferenceView.html" title="class in MapOptics">ReferenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../MapOptics/SVView.html#getChosenSV()" class="member-name-link">getChosenSV()</a> - Method in class MapOptics.<a href="../MapOptics/SVView.html" title="class in MapOptics">SVView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getChr()" class="member-name-link">getChr()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets the name of the chromosome corresponding to this gene</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getChr1()" class="member-name-link">getChr1()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets chromosome 1</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getChr2()" class="member-name-link">getChr2()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets chromosome 2 data</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getChrNames()" class="member-name-link">getChrNames()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets names of chromosomes in karyotype</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getChromosomes()" class="member-name-link">getChromosomes()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets all the chromosomes of this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getChrRelativeStarts()" class="member-name-link">getChrRelativeStarts()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets chromosomes relative start</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getChrSizes()" class="member-name-link">getChrSizes()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets list of size of chromosomes in karyotype</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getChrSizesCircos()" class="member-name-link">getChrSizesCircos()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets chromosomes sites</div>
</dd>
<dt><a href="../Algorithms/Variants.html#getCigQrySites()" class="member-name-link">getCigQrySites()</a> - Method in class Algorithms.<a href="../Algorithms/Variants.html" title="class in Algorithms">Variants</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Cigar.html#getCigQrySites()" class="member-name-link">getCigQrySites()</a> - Method in class DataTypes.<a href="../DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></dt>
<dd>
<div class="block">Get query cigar sites</div>
</dd>
<dt><a href="../Algorithms/Variants.html#getCigRefSites()" class="member-name-link">getCigRefSites()</a> - Method in class Algorithms.<a href="../Algorithms/Variants.html" title="class in Algorithms">Variants</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Cigar.html#getCigRefSites()" class="member-name-link">getCigRefSites()</a> - Method in class DataTypes.<a href="../DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></dt>
<dd>
<div class="block">Get reference cigar sites</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getCmap()" class="member-name-link">getCmap()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets the cmap corresponding to this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getCmapByID(java.lang.Integer)" class="member-name-link">getCmapByID(Integer)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Returns the data of a contig of a given ID from this Cmap, for use when
 accessing CmapData for an Xmap file.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getCmapBySize(java.lang.Double)" class="member-name-link">getCmapBySize(Double)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Returns the data of a contig of a given size from this Cmap, for use when
 assigning a chromosome name using the Karyotype file.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getCmapID()" class="member-name-link">getCmapID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets cmap ID associated with the chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html#getCmapID()" class="member-name-link">getCmapID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Match</a></dt>
<dd>
<div class="block">Gets cmap id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getCmapID()" class="member-name-link">getCmapID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getCmapIDs()" class="member-name-link">getCmapIDs()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets all the contigs ids within this cmap file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getCmapIDs()" class="member-name-link">getCmapIDs()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets the IDs of the cmaps of this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getCmapQry()" class="member-name-link">getCmapQry()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">gets query cmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getCmapRef()" class="member-name-link">getCmapRef()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Gets reference cmap data</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html#getColumnClass(int)" class="member-name-link">getColumnClass(int)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a></dt>
<dd>
<div class="block">Gets class of given column</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html#getColumnCount()" class="member-name-link">getColumnCount()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a></dt>
<dd>
<div class="block">Gets number of columns</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html#getColumnName(int)" class="member-name-link">getColumnName(int)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a></dt>
<dd>
<div class="block">Gets a column name by index</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getConfidence()" class="member-name-link">getConfidence()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap confidence data</div>
</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getConfidence()" class="member-name-link">getConfidence()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets confidence score</div>
</dd>
<dt><a href="../DataTypes/Query.html#getConfidence()" class="member-name-link">getConfidence()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getConfidence()" class="member-name-link">getConfidence()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/SSH.html#getConnection()" class="member-name-link">getConnection()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/SSH.html" title="class in ComparativeGenomics.ServerHandling">SSH</a></dt>
<dd>
<div class="block">Gets the boolean indicating if the server is connected via SSH or not.</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#getConnection()" class="member-name-link">getConnection()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/RefContig.html#getConnections()" class="member-name-link">getConnections()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/ContigInfo.html#getContigLen()" class="member-name-link">getContigLen()</a> - Method in class DataTypes.<a href="../DataTypes/ContigInfo.html" title="class in DataTypes">ContigInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getCoverage()" class="member-name-link">getCoverage()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/LabelInfo.html#getCoverage()" class="member-name-link">getCoverage()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getCoverage()" class="member-name-link">getCoverage()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getDelQryIDs()" class="member-name-link">getDelQryIDs()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getDensities()" class="member-name-link">getDensities()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets densities</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getDensity()" class="member-name-link">getDensity()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getDirection1()" class="member-name-link">getDirection1()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets reference direction 1</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getDirection2()" class="member-name-link">getDirection2()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets reference direction 2</div>
</dd>
<dt><a href="../DataTypes/Inversion.html#getDistances()" class="member-name-link">getDistances()</a> - Static method in class DataTypes.<a href="../DataTypes/Inversion.html" title="class in DataTypes">Inversion</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">Gets distances</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets the end position of this gene</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/QueryPanel.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></dt>
<dd>
<div class="block">Gets end of the query</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets end</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getEnz()" class="member-name-link">getEnz()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the restriction enzyme used in this job</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Annot.html#getFeatureByChr(java.lang.String)" class="member-name-link">getFeatureByChr(String)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Annot.html" title="class in ComparativeGenomics.FileHandling">Annot</a></dt>
<dd>
<div class="block">Accesses all of the Gene objects associated with that chromosome from the chrAnnotations HashMap.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getFileName()" class="member-name-link">getFileName()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets this cmap file name</div>
</dd>
<dt><a href="../DataTypes/Query.html#getFirstAlignment()" class="member-name-link">getFirstAlignment()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/ConflictFileReader.html#getFirstRows()" class="member-name-link">getFirstRows()</a> - Static method in class FileHandling.<a href="../FileHandling/ConflictFileReader.html" title="class in FileHandling">ConflictFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getGCAcrossSequence()" class="member-name-link">getGCAcrossSequence()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">GC% in windows across this chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html#getGCContent()" class="member-name-link">getGCContent()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></dt>
<dd>
<div class="block">Gets GC content</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/MapOpticsRectangle.html#getGene()" class="member-name-link">getGene()</a> - Method in class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/MapOpticsRectangle.html" title="class in ComparativeGenomics.Drawing">MapOpticsRectangle</a></dt>
<dd>
<div class="block">Returns the gene associated with this rectangle</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getGeneFusion()" class="member-name-link">getGeneFusion()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets number of genes merged by this breakpoint</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Annot.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Annot.html" title="class in ComparativeGenomics.FileHandling">Annot</a></dt>
<dd>
<div class="block">Gets genes of the annotation file</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/QueryPanel.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></dt>
<dd>
<div class="block">Gets genes within the range</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getGenomeSize()" class="member-name-link">getGenomeSize()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets genome size</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getHitEnum()" class="member-name-link">getHitEnum()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap hit enum data</div>
</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getHitEnum()" class="member-name-link">getHitEnum()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets hit enum</div>
</dd>
<dt><a href="../DataTypes/Query.html#getHitEnum()" class="member-name-link">getHitEnum()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ExternalServer.html#getHost()" class="member-name-link">getHost()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></dt>
<dd>
<div class="block">Sets this servs host address</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#gethScale()" class="member-name-link">gethScale()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html#getID()" class="member-name-link">getID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></dt>
<dd>
<div class="block">Gets this cmap ID</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getID()" class="member-name-link">getID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap id</div>
</dd>
<dt><a href="../DataTypes/Query.html#getID()" class="member-name-link">getID()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getIds()" class="member-name-link">getIds()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets ID of contigs supporting this breakpoint</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getImageModel()" class="member-name-link">getImageModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/DetectSV.html#getIndels()" class="member-name-link">getIndels()</a> - Method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getIndels()" class="member-name-link">getIndels()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Get all the indels of this chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getIndels()" class="member-name-link">getIndels()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap indels</div>
</dd>
<dt><a href="../MapOptics/SVView.html#getIndels()" class="member-name-link">getIndels()</a> - Static method in class MapOptics.<a href="../MapOptics/SVView.html" title="class in MapOptics">SVView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Indel.html#getIndels(java.util.Map,java.util.Map,java.util.Map)" class="member-name-link">getIndels(Map&lt;Integer, List&lt;Integer&gt;&gt;, Map&lt;Integer, Double&gt;, Map&lt;Integer, Double&gt;)</a> - Static method in class DataTypes.<a href="../DataTypes/Indel.html" title="class in DataTypes">Indel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Drawing/Curve2D.html#getInfo()" class="member-name-link">getInfo()</a> - Method in class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/Curve2D.html" title="class in ComparativeGenomics.Drawing">Curve2D</a></dt>
<dd>
<div class="block">Gets translocations information for future user interactivity</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getInfo()" class="member-name-link">getInfo()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets supplementary information on this gene</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getInfo()" class="member-name-link">getInfo()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets chromosomes sizes and names</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MyChartRenderer.html#getItemPaint(int,int)" class="member-name-link">getItemPaint(int, int)</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MyChartRenderer.html" title="class in UserInterface.ModelsAndRenderers">MyChartRenderer</a></dt>
<dd>
<div class="block">Gets item to paint</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/JsonFiles.html#getJobsFromJson(java.util.List)" class="member-name-link">getJobsFromJson(List&lt;Job&gt;)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/JsonFiles.html" title="class in ComparativeGenomics.FileHandling">JsonFiles</a></dt>
<dd>
<div class="block">Gets jobs saved in this jobs json file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getKaryotype()" class="member-name-link">getKaryotype()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets the karyotype corresponding to this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getLabelChannel()" class="member-name-link">getLabelChannel()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getLabelChannel()" class="member-name-link">getLabelChannel()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets channel label</div>
</dd>
<dt><a href="../DataTypes/LabelInfo.html#getLabelChannel()" class="member-name-link">getLabelChannel()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getLabelChannels()" class="member-name-link">getLabelChannels()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Labels channels of this xmap file</div>
</dd>
<dt><a href="../DataTypes/ContigInfo.html#getLabelInfo()" class="member-name-link">getLabelInfo()</a> - Method in class DataTypes.<a href="../DataTypes/ContigInfo.html" title="class in DataTypes">ContigInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getLabelModel()" class="member-name-link">getLabelModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/LabelInfo.html#getLabelPos()" class="member-name-link">getLabelPos()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/QryContig.html#getLabels()" class="member-name-link">getLabels()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getLabels()" class="member-name-link">getLabels()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/RefContig.html#getLabels()" class="member-name-link">getLabels()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getLabels()" class="member-name-link">getLabels()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html#getLength()" class="member-name-link">getLength()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></dt>
<dd>
<div class="block">Gets the length of the reference covered by this cmap ID</div>
</dd>
<dt><a href="../DataTypes/Query.html#getLength()" class="member-name-link">getLength()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getLength()" class="member-name-link">getLength()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getLengths()" class="member-name-link">getLengths()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets lengths</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getLinkID()" class="member-name-link">getLinkID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this smap link id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getLocalisedTranslocations(int,int)" class="member-name-link">getLocalisedTranslocations(int, int)</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Get translocations according to their position's in the array list of
 translocation</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html#getMatch()" class="member-name-link">getMatch()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Match</a></dt>
<dd>
<div class="block">Gets match</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getMatchCount()" class="member-name-link">getMatchCount()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getMatches()" class="member-name-link">getMatches()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getMatchesByXmapID(java.lang.Integer)" class="member-name-link">getMatchesByXmapID(Integer)</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Get the names of the chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets the name of this gene</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets this genome name</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Enzyme.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Enzyme.html" title="class in ComparativeGenomics.ServerHandling">Enzyme</a></dt>
<dd>
<div class="block">Gets this enzymes name</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ExternalServer.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></dt>
<dd>
<div class="block">Sets this servers name</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getName()" class="member-name-link">getName()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets this jobs name</div>
</dd>
<dt><a href="../DataTypes/RefContig.html#getName()" class="member-name-link">getName()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getNickase()" class="member-name-link">getNickase()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets digestion enzyme used to digest this cmap file</div>
</dd>
<dt><a href="../FileHandling/CmapReader.html#getNickaseEnzyme(java.lang.String)" class="member-name-link">getNickaseEnzyme(String)</a> - Method in class FileHandling.<a href="../FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></dt>
<dd>
<div class="block">Extract nickase enzyme from file containing enzyme information.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html#getNs()" class="member-name-link">getNs()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></dt>
<dd>
<div class="block">Gets gaps</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html#getNumChrs()" class="member-name-link">getNumChrs()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></dt>
<dd>
<div class="block">Gets the number of chromosomes in this genome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getNumChrs()" class="member-name-link">getNumChrs()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Gets number of chromosome in karyotype</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getNumConsensus()" class="member-name-link">getNumConsensus()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets number of contigs contained within this cmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getNumDuplications()" class="member-name-link">getNumDuplications()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets number of duplications in chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Annot.html#getNumFeaturesByChr(java.lang.String)" class="member-name-link">getNumFeaturesByChr(String)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Annot.html" title="class in ComparativeGenomics.FileHandling">Annot</a></dt>
<dd>
<div class="block">Returns the number of genes for each chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getNumIndels()" class="member-name-link">getNumIndels()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets number of indels in chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getNumInversions()" class="member-name-link">getNumInversions()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets number of inversions in chromosome</div>
</dd>
<dt><a href="../DataTypes/Query.html#getNumMatches()" class="member-name-link">getNumMatches()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getNumSupports()" class="member-name-link">getNumSupports()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets number of genes supporting this breakpoint</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getNumTranslocations()" class="member-name-link">getNumTranslocations()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets the number of translocations in chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getOccurance()" class="member-name-link">getOccurance()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/LabelInfo.html#getOccurance()" class="member-name-link">getOccurance()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getOri()" class="member-name-link">getOri()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap orientation data</div>
</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getOrientation()" class="member-name-link">getOrientation()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets contig orientation</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getOrientation()" class="member-name-link">getOrientation()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getOrientation()" class="member-name-link">getOrientation()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getOrientation()" class="member-name-link">getOrientation()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/CalculateOverlaps.html#getOverlapRegions(DataTypes.Reference)" class="member-name-link">getOverlapRegions(Reference)</a> - Static method in class Algorithms.<a href="../Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></dt>
<dd>
<div class="block">Get overlapping regions</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getOverlaps()" class="member-name-link">getOverlaps()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ExternalServer.html#getPassword()" class="member-name-link">getPassword()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></dt>
<dd>
<div class="block">Sets this servers password</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html#getPercentNs()" class="member-name-link">getPercentNs()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></dt>
<dd>
<div class="block">Gets percentage of Ns</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getPipeline()" class="member-name-link">getPipeline()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the alignment algorithm used to execute this job</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getPosition()" class="member-name-link">getPosition()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getPotentialTranslocations()" class="member-name-link">getPotentialTranslocations()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets potential translocations</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html#getQry()" class="member-name-link">getQry()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Pair</a></dt>
<dd>
<div class="block">Gets query</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getQry()" class="member-name-link">getQry()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the query genome file path of this job</div>
</dd>
<dt><a href="../Datasets/UserEdited/SearchRegionData.html#getQry()" class="member-name-link">getQry()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SearchRegionData.html" title="class in Datasets.UserEdited">SearchRegionData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getQryAlignEnd()" class="member-name-link">getQryAlignEnd()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets query alignment end</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getQryAlignEnd()" class="member-name-link">getQryAlignEnd()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getQryAlignStart()" class="member-name-link">getQryAlignStart()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets query alignment start</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getQryAlignStart()" class="member-name-link">getQryAlignStart()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getQryAnnot()" class="member-name-link">getQryAnnot()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the query annotation file path</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getQryCmap()" class="member-name-link">getQryCmap()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets query cmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html#getQryCmapID()" class="member-name-link">getQryCmapID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Match</a></dt>
<dd>
<div class="block">Gets query cmap id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getQryCmaps()" class="member-name-link">getQryCmaps()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets all of the CmapData objects associated with this query Cmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getQryCmapsByID(java.lang.Integer)" class="member-name-link">getQryCmapsByID(Integer)</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets a query CmapData object by ID</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html#getQryCmapSite()" class="member-name-link">getQryCmapSite()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Match.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Match</a></dt>
<dd>
<div class="block">Gets query cmap site</div>
</dd>
<dt><a href="../Datasets/UserEdited/SavedQryData.html#getQryConnections(java.lang.String)" class="member-name-link">getQryConnections(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserQryData.html#getQryConnections(java.lang.String)" class="member-name-link">getQryConnections(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getQryContigID()" class="member-name-link">getQryContigID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this query contig id</div>
</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getQryContigs()" class="member-name-link">getQryContigs()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getQryContigs(java.lang.String)" class="member-name-link">getQryContigs(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getQryEnd()" class="member-name-link">getQryEnd()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets query end</div>
</dd>
<dt><a href="../DataTypes/SV.html#getQryEndPos()" class="member-name-link">getQryEndPos()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getQryEndSite()" class="member-name-link">getQryEndSite()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getQryFile()" class="member-name-link">getQryFile()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets query file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getQryID()" class="member-name-link">getQryID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap query id</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getQryID()" class="member-name-link">getQryID()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets query ID</div>
</dd>
<dt><a href="../DataTypes/SV.html#getQryID()" class="member-name-link">getQryID()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getQryLen()" class="member-name-link">getQryLen()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap query length data</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#getQryLen()" class="member-name-link">getQryLen()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getQryModel()" class="member-name-link">getQryModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getQryOrg()" class="member-name-link">getQryOrg()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the query species</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getQryPos()" class="member-name-link">getQryPos()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this query position</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html#getQrySite()" class="member-name-link">getQrySite()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Pair</a></dt>
<dd>
<div class="block">Gets query site</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getQryStart()" class="member-name-link">getQryStart()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets query start</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#getQryStart()" class="member-name-link">getQryStart()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getQryStartPos()" class="member-name-link">getQryStartPos()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getQryStartSite()" class="member-name-link">getQryStartSite()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getQryViewRect()" class="member-name-link">getQryViewRect()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getQryViewRect()" class="member-name-link">getQryViewRect()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getQryViewRefModel()" class="member-name-link">getQryViewRefModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getQryViewSites()" class="member-name-link">getQryViewSites()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RefViewData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RefViewData.html" title="class in Datasets.Default">RefViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/SummaryViewData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/SummaryViewData.html" title="class in Datasets.Default">SummaryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedQryData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedRefData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedRefData.html" title="class in Datasets.UserEdited">SavedRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserQryData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserRefData.html#getQueries()" class="member-name-link">getQueries()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserRefData.html" title="class in Datasets.UserEdited">UserRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getQueries()" class="member-name-link">getQueries()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RefViewData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RefViewData.html" title="class in Datasets.Default">RefViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/SummaryViewData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/SummaryViewData.html" title="class in Datasets.Default">SummaryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedQryData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedRefData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedRefData.html" title="class in Datasets.UserEdited">SavedRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserQryData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserRefData.html#getQueries(java.lang.String)" class="member-name-link">getQueries(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserRefData.html" title="class in Datasets.UserEdited">UserRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getQueriesMappedToRef()" class="member-name-link">getQueriesMappedToRef()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets queries mapped to reference</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getQuery(java.lang.String)" class="member-name-link">getQuery(String)</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/XmapReader.html#getQueryData(UserInterface.ModelsAndRenderers.MapOpticsModel,java.lang.String)" class="member-name-link">getQueryData(MapOpticsModel, String)</a> - Static method in class FileHandling.<a href="../FileHandling/XmapReader.html" title="class in FileHandling">XmapReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getQueryIDs()" class="member-name-link">getQueryIDs()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getQueryLen()" class="member-name-link">getQueryLen()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets length of queries</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getQueryList()" class="member-name-link">getQueryList()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets list of queries</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/QueryPanel.html#getRange()" class="member-name-link">getRange()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></dt>
<dd>
<div class="block">Gets size of the query</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/QueryShape.html#getRect()" class="member-name-link">getRect()</a> - Method in class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/QueryShape.html" title="class in ComparativeGenomics.Drawing">QueryShape</a></dt>
<dd>
<div class="block">Gets rectangle</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/XmapShape.html#getRect()" class="member-name-link">getRect()</a> - Method in class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/XmapShape.html" title="class in ComparativeGenomics.Drawing">XmapShape</a></dt>
<dd>
<div class="block">Gets <code>MapOpticsRectangle</code></div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getRectangle()" class="member-name-link">getRectangle()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getRectangle()" class="member-name-link">getRectangle()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/RefContig.html#getRectangle()" class="member-name-link">getRectangle()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getRectangle()" class="member-name-link">getRectangle()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getRectangleTotalHeight()" class="member-name-link">getRectangleTotalHeight()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets rectangle total height</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getRectangleTotalWidth()" class="member-name-link">getRectangleTotalWidth()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets rectangle width</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html#getRef()" class="member-name-link">getRef()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Pair</a></dt>
<dd>
<div class="block">Gets reference</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getRef()" class="member-name-link">getRef()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the reference genome file of this job</div>
</dd>
<dt><a href="../Datasets/UserEdited/SearchRegionData.html#getRef()" class="member-name-link">getRef()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SearchRegionData.html" title="class in Datasets.UserEdited">SearchRegionData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getRefAlignEnd()" class="member-name-link">getRefAlignEnd()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets reference alignment end</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getRefAlignEnd()" class="member-name-link">getRefAlignEnd()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getRefAlignPos(java.lang.String)" class="member-name-link">getRefAlignPos(String)</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/AlignmentInfo.html#getRefAlignStart()" class="member-name-link">getRefAlignStart()</a> - Method in class DataTypes.<a href="../DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></dt>
<dd>
<div class="block">Gets reference alignment start</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getRefAlignStart()" class="member-name-link">getRefAlignStart()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getRefAnnot()" class="member-name-link">getRefAnnot()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the reference annotation file path</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Translocation.html#getRefChr1Name()" class="member-name-link">getRefChr1Name()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Translocation.html" title="class in ComparativeGenomics.StructuralVariant">Translocation</a></dt>
<dd>
<div class="block">Gets first chromosome name</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Translocation.html#getRefChr2Name()" class="member-name-link">getRefChr2Name()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Translocation.html" title="class in ComparativeGenomics.StructuralVariant">Translocation</a></dt>
<dd>
<div class="block">Gets second chromosome name</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getRefCmap()" class="member-name-link">getRefCmap()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets reference cmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getRefCmapData()" class="member-name-link">getRefCmapData()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets the CmapData associated with this chromosome</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getRefContigID()" class="member-name-link">getRefContigID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this reference contig id</div>
</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getRefContigs()" class="member-name-link">getRefContigs()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getRefContigs(java.lang.String)" class="member-name-link">getRefContigs(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getRefDensity()" class="member-name-link">getRefDensity()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getRefEnd()" class="member-name-link">getRefEnd()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets reference end</div>
</dd>
<dt><a href="../DataTypes/SV.html#getRefEndPos()" class="member-name-link">getRefEndPos()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Indel.html#getRefEndSite()" class="member-name-link">getRefEndSite()</a> - Method in class DataTypes.<a href="../DataTypes/Indel.html" title="class in DataTypes">Indel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getRefEndSite()" class="member-name-link">getRefEndSite()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/CmapReader.html#getReferenceData(java.io.File,java.util.Map)" class="member-name-link">getReferenceData(File, Map&lt;Integer, Query&gt;)</a> - Static method in class FileHandling.<a href="../FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></dt>
<dd>
<div class="block">Gets reference data.</div>
</dd>
<dt><a href="../FileHandling/XmapReader.html#getReferenceData(UserInterface.ModelsAndRenderers.MapOpticsModel)" class="member-name-link">getReferenceData(MapOpticsModel)</a> - Static method in class FileHandling.<a href="../FileHandling/XmapReader.html" title="class in FileHandling">XmapReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RefViewData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RefViewData.html" title="class in Datasets.Default">RefViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/SummaryViewData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/SummaryViewData.html" title="class in Datasets.Default">SummaryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedQryData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedRefData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedRefData.html" title="class in Datasets.UserEdited">SavedRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserQryData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserRefData.html#getReferences()" class="member-name-link">getReferences()</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserRefData.html" title="class in Datasets.UserEdited">UserRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getReferences()" class="member-name-link">getReferences()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets references</div>
</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/Default/RefViewData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RefViewData.html" title="class in Datasets.Default">RefViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/Default/SummaryViewData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/SummaryViewData.html" title="class in Datasets.Default">SummaryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedQryData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/SavedRefData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/SavedRefData.html" title="class in Datasets.UserEdited">SavedRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserQryData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../Datasets/UserEdited/UserRefData.html#getReferences(java.lang.String)" class="member-name-link">getReferences(String)</a> - Static method in class Datasets.UserEdited.<a href="../Datasets/UserEdited/UserRefData.html" title="class in Datasets.UserEdited">UserRefData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getRefFile()" class="member-name-link">getRefFile()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gest reference file</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getRefGenome()" class="member-name-link">getRefGenome()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Gets reference genome data</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getRefGenomeName()" class="member-name-link">getRefGenomeName()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Gets reference genome name</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getRefID()" class="member-name-link">getRefID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap reference id</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getRefID()" class="member-name-link">getRefID()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets reference ID</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getRefID()" class="member-name-link">getRefID()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getRefID1()" class="member-name-link">getRefID1()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getRefID2()" class="member-name-link">getRefID2()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getRefLen()" class="member-name-link">getRefLen()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets this xmap reference length data</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#getRefLen()" class="member-name-link">getRefLen()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getRefLen()" class="member-name-link">getRefLen()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets length of referecnes</div>
</dd>
<dt><a href="../Datasets/Default/RawFileData.html#getRefLengths()" class="member-name-link">getRefLengths()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getRefModel()" class="member-name-link">getRefModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getRefOrg()" class="member-name-link">getRefOrg()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets the reference species</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getRefPos()" class="member-name-link">getRefPos()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this reference position</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getRefPos1()" class="member-name-link">getRefPos1()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets reference breakpoint position 1</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getRefPos2()" class="member-name-link">getRefPos2()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gets reference breakpoint position 2</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html#getRefSite()" class="member-name-link">getRefSite()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Pair.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Pair</a></dt>
<dd>
<div class="block">Gets reference site</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getRefSites()" class="member-name-link">getRefSites()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets the hashmap containing all the sites within this chromosome, with the
 key being the site ID and the value being the site object</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html#getRefStart()" class="member-name-link">getRefStart()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></dt>
<dd>
<div class="block">Gets reference start</div>
</dd>
<dt><a href="../Datasets/Default/QueryViewData.html#getRefStart()" class="member-name-link">getRefStart()</a> - Static method in class Datasets.Default.<a href="../Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getRefStartPos()" class="member-name-link">getRefStartPos()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Indel.html#getRefStartSite()" class="member-name-link">getRefStartSite()</a> - Method in class DataTypes.<a href="../DataTypes/Indel.html" title="class in DataTypes">Indel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getRefStartSite()" class="member-name-link">getRefStartSite()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getRefViewOffsetX()" class="member-name-link">getRefViewOffsetX()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getRefViewOffsetY()" class="member-name-link">getRefViewOffsetY()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getRefViewRect()" class="member-name-link">getRefViewRect()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getRefViewRect()" class="member-name-link">getRefViewRect()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Query.html#getRefViewSites()" class="member-name-link">getRefViewSites()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getRegions()" class="member-name-link">getRegions()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/BestEnzyme.html#getResult()" class="member-name-link">getResult()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/BestEnzyme.html" title="class in ComparativeGenomics.FileHandling">BestEnzyme</a></dt>
<dd>
<div class="block">Return HashMap of results</div>
</dd>
<dt><a href="../DataTypes/Cigar.html#getRevPalindromes(java.util.List,java.util.List)" class="member-name-link">getRevPalindromes(List&lt;String&gt;, List&lt;String&gt;)</a> - Method in class DataTypes.<a href="../DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></dt>
<dd>
<div class="block">Find sequences which are palindromic within the CIGAR string</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html#getRowCount()" class="member-name-link">getRowCount()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a></dt>
<dd>
<div class="block">Gets number of rows</div>
</dd>
<dt><a href="../DataTypes/Reference.html#getSavedDelQryIDs()" class="member-name-link">getSavedDelQryIDs()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getSelected()" class="member-name-link">getSelected()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets this gene selection information</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getSelectedRef()" class="member-name-link">getSelectedRef()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets selected reference</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getSelectedRefID()" class="member-name-link">getSelectedRefID()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets selected reference id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html#getSeq()" class="member-name-link">getSeq()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></dt>
<dd>
<div class="block">Gets sequence</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Access a specific sequence from the sequences HashMap by sequence size</div>
</dd>
<dt><a href="../DataTypes/QryContig.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class DataTypes.<a href="../DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/RefContig.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class DataTypes.<a href="../DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Fasta.html#getSequence(java.lang.Double)" class="member-name-link">getSequence(Double)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Fasta.html" title="class in ComparativeGenomics.FileHandling">Fasta</a></dt>
<dd>
<div class="block">Gets fasta sequence corresponding to a given size</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getServer()" class="member-name-link">getServer()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Get the information of the server on which this job is run</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/JsonFiles.html#getServersFromJson(java.util.List)" class="member-name-link">getServersFromJson(List&lt;ExternalServer&gt;)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/JsonFiles.html" title="class in ComparativeGenomics.FileHandling">JsonFiles</a></dt>
<dd>
<div class="block">Gets the servers from this servers json file</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Enzyme.html#getSite()" class="member-name-link">getSite()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Enzyme.html" title="class in ComparativeGenomics.ServerHandling">Enzyme</a></dt>
<dd>
<div class="block">Gets this enzymes site</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html#getSite(java.lang.Integer)" class="member-name-link">getSite(Integer)</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></dt>
<dd>
<div class="block">Gets a <code>Site</code> object associated with a cmap ID</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getSiteID()" class="member-name-link">getSiteID()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html#getSites()" class="member-name-link">getSites()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></dt>
<dd>
<div class="block">Gets all the sites associated with a cmap ID</div>
</dd>
<dt><a href="../DataTypes/Query.html#getSites()" class="member-name-link">getSites()</a> - Method in class DataTypes.<a href="../DataTypes/Query.html" title="class in DataTypes">Query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Reference.html#getSites()" class="member-name-link">getSites()</a> - Method in class DataTypes.<a href="../DataTypes/Reference.html" title="class in DataTypes">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/Inversion.html#getSites(java.lang.String,java.util.Map,java.util.Map,java.util.Map,java.util.Map)" class="member-name-link">getSites(String, Map&lt;Integer, List&lt;Integer&gt;&gt;, Map&lt;Integer, List&lt;Integer&gt;&gt;, Map&lt;Integer, Double&gt;, Map&lt;Integer, Double&gt;)</a> - Static method in class DataTypes.<a href="../DataTypes/Inversion.html" title="class in DataTypes">Inversion</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">Matches the distance sites between adjacent label sites on ref and
 those between adjacent labels on the reversed query</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html#getSize()" class="member-name-link">getSize()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></dt>
<dd>
<div class="block">Gets chromosome size</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html#getSize()" class="member-name-link">getSize()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></dt>
<dd>
<div class="block">Gets size</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getSize()" class="member-name-link">getSize()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets size</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Smap.html#getSmapFormat()" class="member-name-link">getSmapFormat()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Smap.html" title="class in ComparativeGenomics.FileHandling">Smap</a></dt>
<dd>
<div class="block">Gets smap format</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getSmapid()" class="member-name-link">getSmapid()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this smap id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Smap.html#getSmapTransloc()" class="member-name-link">getSmapTransloc()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Smap.html" title="class in ComparativeGenomics.FileHandling">Smap</a></dt>
<dd>
<div class="block">Gets translocations from RefAligner smap output</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getSource()" class="member-name-link">getSource()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets this gene source</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getStart()" class="member-name-link">getStart()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets the start position of this gene</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/QueryPanel.html#getStart()" class="member-name-link">getStart()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></dt>
<dd>
<div class="block">Gets start of the query</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getStart()" class="member-name-link">getStart()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets starts</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/Job.html#getStatus()" class="member-name-link">getStatus()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></dt>
<dd>
<div class="block">Gets job status</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html#getStdDev()" class="member-name-link">getStdDev()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/LabelInfo.html#getStdDev()" class="member-name-link">getStdDev()</a> - Method in class DataTypes.<a href="../DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/Panels/EnzymeChartPanel.html#getStyle()" class="member-name-link">getStyle()</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/EnzymeChartPanel.html" title="class in ComparativeGenomics.Panels">EnzymeChartPanel</a></dt>
<dd>
<div class="block">Gets this plots style</div>
</dd>
<dt><a href="../MapOptics/SVView.html#getStyle()" class="member-name-link">getStyle()</a> - Static method in class MapOptics.<a href="../MapOptics/SVView.html" title="class in MapOptics">SVView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/XmapReader.html#getSummaryData(java.io.File,boolean)" class="member-name-link">getSummaryData(File, boolean)</a> - Static method in class FileHandling.<a href="../FileHandling/XmapReader.html" title="class in FileHandling">XmapReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../FileHandling/CmapReader.html#getSummaryData(java.io.File,java.util.Map)" class="member-name-link">getSummaryData(File, Map&lt;Integer, Reference&gt;)</a> - Static method in class FileHandling.<a href="../FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></dt>
<dd>
<div class="block">Gets summary data.</div>
</dd>
<dt><a href="../Algorithms/DetectSV.html#getSV(java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">getSV(String, String, String, String, String)</a> - Method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../Algorithms/DetectSV.html#getSVList()" class="member-name-link">getSVList()</a> - Method in class Algorithms.<a href="../Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/TableModels.html#getSVModel()" class="member-name-link">getSVModel()</a> - Static method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../DataTypes/SV.html#getSVSize()" class="member-name-link">getSVSize()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/EditableHeaderRenderer.html#getTableCellRendererComponent(javax.swing.JTable,java.lang.Object,boolean,boolean,int,int)" class="member-name-link">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/EditableHeaderRenderer.html" title="class in UserInterface.ModelsAndRenderers">EditableHeaderRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getTranslocations()" class="member-name-link">getTranslocations()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Gets translocations data</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Smap.html#getTxtTransloc()" class="member-name-link">getTxtTransloc()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Smap.html" title="class in ComparativeGenomics.FileHandling">Smap</a></dt>
<dd>
<div class="block">Get translocations from FaNDOM text file output</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html#getType()" class="member-name-link">getType()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></dt>
<dd>
<div class="block">Gets this gene type</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html#getType()" class="member-name-link">getType()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></dt>
<dd>
<div class="block">Gest SV type</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getType()" class="member-name-link">getType()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this type</div>
</dd>
<dt><a href="../ComparativeGenomics/StructuralVariant/Indel.html#getType()" class="member-name-link">getType()</a> - Method in class ComparativeGenomics.StructuralVariant.<a href="../ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></dt>
<dd>
<div class="block">Gets type</div>
</dd>
<dt><a href="../DataTypes/SV.html#getType()" class="member-name-link">getType()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ExternalServer.html#getUser()" class="member-name-link">getUser()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></dt>
<dd>
<div class="block">sets this servers username</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Annot.html#getValidity()" class="member-name-link">getValidity()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Annot.html" title="class in ComparativeGenomics.FileHandling">Annot</a></dt>
<dd>
<div class="block">Indicates if A valid file exists, is in proper format and is not empty</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getValidity()" class="member-name-link">getValidity()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Checks cmap validity (file is in cmap format, exists, is not null)</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Karyotype.html#getValidity()" class="member-name-link">getValidity()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></dt>
<dd>
<div class="block">Indicates file validity.</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getValidity()" class="member-name-link">getValidity()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Returns the validity of the file</div>
</dd>
<dt><a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html#getValueAt(int,int)" class="member-name-link">getValueAt(int, int)</a> - Method in class ComparativeGenomics.Panels.<a href="../ComparativeGenomics/Panels/CompGenStart.JobTableModel.html" title="class in ComparativeGenomics.Panels">CompGenStart.JobTableModel</a></dt>
<dd>
<div class="block">Gets value at given column and row</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Cmap.html#getVersion()" class="member-name-link">getVersion()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></dt>
<dd>
<div class="block">Gets this cmap file version</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getVersion()" class="member-name-link">getVersion()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets version of this xmap file</div>
</dd>
<dt><a href="../ComparativeGenomics/ServerHandling/ExternalServer.html#getWorkingDir()" class="member-name-link">getWorkingDir()</a> - Method in class ComparativeGenomics.ServerHandling.<a href="../ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></dt>
<dd>
<div class="block">Sets this servers working directory</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html#getXmap()" class="member-name-link">getXmap()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></dt>
<dd>
<div class="block">Gets xmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getXmap()" class="member-name-link">getXmap()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets xmap</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getXmap(java.lang.Integer)" class="member-name-link">getXmap(Integer)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getXmapByQry()" class="member-name-link">getXmapByQry()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Returns a Hashmap of all maps (values) corresponding to a query contig ID (key) in the XMAP</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getXmapByRef()" class="member-name-link">getXmapByRef()</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Returns a Hashmap of all maps (values) corresponding to a reference contig ID (key) in the XMAP</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/Xmap.html#getXmapByXmapID(java.lang.Integer)" class="member-name-link">getXmapByXmapID(Integer)</a> - Method in class ComparativeGenomics.FileHandling.<a href="../ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></dt>
<dd>
<div class="block">Gets xmap by their IDs</div>
</dd>
<dt><a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html#getXmapFile()" class="member-name-link">getXmapFile()</a> - Method in class UserInterface.ModelsAndRenderers.<a href="../UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></dt>
<dd>
<div class="block">Gets xmap file</div>
</dd>
<dt><a href="../ComparativeGenomics/Drawing/XmapShape.html#getXmapID()" class="member-name-link">getXmapID()</a> - Method in class ComparativeGenomics.Drawing.<a href="../ComparativeGenomics/Drawing/XmapShape.html" title="class in ComparativeGenomics.Drawing">XmapShape</a></dt>
<dd>
<div class="block">Gets xmap id</div>
</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getXmapID1()" class="member-name-link">getXmapID1()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this xmap 1 id</div>
</dd>
<dt><a href="../DataTypes/SV.html#getXmapID1()" class="member-name-link">getXmapID1()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html#getXmapID2()" class="member-name-link">getXmapID2()</a> - Method in class ComparativeGenomics.FileHandling.DataHandling.<a href="../ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></dt>
<dd>
<div class="block">Gets this xmap 2 id</div>
</dd>
<dt><a href="../DataTypes/SV.html#getXmapID2()" class="member-name-link">getXmapID2()</a> - Method in class DataTypes.<a href="../DataTypes/SV.html" title="class in DataTypes">SV</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
