<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (19) on Tue Aug 22 11:59:40 BST 2023 -->
<title>All Classes and Interfaces</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2023-08-22">
<meta name="description" content="class index">
<meta name="generator" content="javadoc/AllClassesIndexWriter">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="script-dir/jquery-3.6.0.min.js"></script>
<script type="text/javascript" src="script-dir/jquery-ui.min.js"></script>
</head>
<body class="all-classes-index-page">
<script type="text/javascript">var pathtoroot = "./";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html#all-classes">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="All Classes and Interfaces" class="title">All Classes and Interfaces</h1>
</div>
<div id="all-classes-table">
<div class="caption"><span>Classes</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Alignment.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Alignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the alignment files produced by the job.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/AlignmentInfo.html" title="class in DataTypes">AlignmentInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores information on alignments</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/AlignmentsOnChromosomeChartPanel.html" title="class in ComparativeGenomics.Panels">AlignmentsOnChromosomeChartPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Plots the number of alignments found per each cmap ID of
 the query genome accross a chromosome from the reference genome using XChart.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/AlignmentsPerChromosomeChartPanel.html" title="class in ComparativeGenomics.Panels">AlignmentsPerChromosomeChartPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Annot.html" title="class in ComparativeGenomics.FileHandling">Annot</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Verifies file format, parse and save the contents of a
 GFF3 (General Feature File) or GTF (General Transfer Format) file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/BestEnzyme.html" title="class in ComparativeGenomics.FileHandling">BestEnzyme</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Read in the file generated from the calc_best_enz.sh script to determine the
 frequency of different digestion enzymes for a given fasta file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Algorithms/CalculateOverlaps.html" title="class in Algorithms">CalculateOverlaps</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculates and saved query contigs which overlap in regions of their alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Chromosome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Chromosome</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Combines the CmapData, XmapData, annotations (gene), structural variants and
 context of the chromosome within a genom</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/ChromosomeChartPanel.html" title="class in ComparativeGenomics.Panels">ChromosomeChartPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Plots the number of each different type of SV event per chromosome within the reference genome using XChart.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/ChromosomePanel.html" title="class in ComparativeGenomics.Panels">ChromosomePanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Extends javax.swing.JPanel which draws all of the XmapData
 objects corresponding to a chosen Chromosome.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Cigar.html" title="class in DataTypes">Cigar</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parse and save cigar content</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/CircosPanel.html" title="class in ComparativeGenomics.Panels">CircosPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Draws the chromosome making up the genome in a circle.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Cmap.html" title="class in ComparativeGenomics.FileHandling">Cmap</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores all of the contiguous maps contained within the
 cmap file the data are stored into hashmaps to allow for efficient access.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/CmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">CmapData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the data corresponding to each cmap ID within a CMAP file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="FileHandling/CmapReader.html" title="class in FileHandling">CmapReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Validates and reads CMAP files</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/CompGenStart.html" title="class in ComparativeGenomics.Panels">CompGenStart</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides the GUI and methods to start an alignment Job on an ExternalServer.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/CompGenView.html" title="class in ComparativeGenomics.Panels">CompGenView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Panel on which alignments of two genomes using optical mapping data are displayed.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="FileHandling/ConflictFileReader.html" title="class in FileHandling">ConflictFileReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="FileHandling/ConflictFileWriter.html" title="class in FileHandling">ConflictFileWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/ContigInfo.html" title="class in DataTypes">ContigInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Drawing/Curve2D.html" title="class in ComparativeGenomics.Drawing">Curve2D</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Draws curves between two chromosomes in circos view to graphically indicate a
 translocation event.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Algorithms/DetectSV.html" title="class in Algorithms">DetectSV</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/StructuralVariant/Duplication.html" title="class in ComparativeGenomics.StructuralVariant">Duplication</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores duplication data</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Duplication.html" title="class in DataTypes">Duplication</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="UserInterface/ModelsAndRenderers/EditableHeaderRenderer.html" title="class in UserInterface.ModelsAndRenderers">EditableHeaderRenderer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="UserInterface/ModelsAndRenderers/EditableHeaderRenderer.MouseEventReposter.html" title="class in UserInterface.ModelsAndRenderers">EditableHeaderRenderer.MouseEventReposter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/ServerHandling/Enzyme.html" title="class in ComparativeGenomics.ServerHandling">Enzyme</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores enzyme information such as name and restriction sites.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/EnzymeChartPanel.html" title="class in ComparativeGenomics.Panels">EnzymeChartPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Plots the results of calc_best_enzymes.sh script of an uploaded fasta file
 using <code>XChart</code></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/ServerHandling/ExternalServer.html" title="class in ComparativeGenomics.ServerHandling">ExternalServer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Saves all of the information about an external server to enable mapoptics to
 connect to it</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Fasta.html" title="class in ComparativeGenomics.FileHandling">Fasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Verifies file format, parse and save the contents of a multi-sequence fasta
 file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="FileHandling/FastaReader.html" title="class in FileHandling">FastaReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reads and parse a FASTA file</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Gene.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Gene</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the gene entries from a GTF or GFFF file via the <code>Annot</code>
 object.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Genome.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Genome</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores all the metadata for example species of the genome,
 amd group the files related to the genome.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/GenomePanel.html" title="class in ComparativeGenomics.Panels">GenomePanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Displays all the chromosomes of the reference.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/StructuralVariant/Indel.html" title="class in ComparativeGenomics.StructuralVariant">Indel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores insertions and deletions alongside any
 associated genes located within the region of the ref genome</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Indel.html" title="class in DataTypes">Indel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/StructuralVariant/Inversion.html" title="class in ComparativeGenomics.StructuralVariant">Inversion</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores inversion data</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Inversion.html" title="class in DataTypes">Inversion</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/ServerHandling/Job.html" title="class in ComparativeGenomics.ServerHandling">Job</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Saves all information related to a submitted job, allowing the data to be downloaded from 
 the server it was run on, and displaying the results.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/JsonFiles.html" title="class in ComparativeGenomics.FileHandling">JsonFiles</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Manages creation and retrieval of information from server Json
 files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Karyotype.html" title="class in ComparativeGenomics.FileHandling">Karyotype</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parses the karyotype file generated from the reference genome produced by
 the run_job.sh script using Samtools faidx into a space separated file with
 the first column being the chromosome.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/LabelInfo.html" title="class in DataTypes">LabelInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="MapOptics/MapOptics.html" title="class in MapOptics">MapOptics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Frame of the application</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Drawing/MapOpticsArc.html" title="class in ComparativeGenomics.Drawing">MapOpticsArc</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Extends <code>Arc2D</code>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="UserInterface/ModelsAndRenderers/MapOpticsModel.html" title="class in UserInterface.ModelsAndRenderers">MapOpticsModel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains data of reference and query, and their alignments.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Drawing/MapOpticsRectangle.html" title="class in ComparativeGenomics.Drawing">MapOpticsRectangle</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Draws gene shapes on <code>QueryPanel</code>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Match.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Match</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the information on the match, so that only the matches related to the
 displayed reference cmap are drawn.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="UserInterface/ModelsAndRenderers/MyChartRenderer.html" title="class in UserInterface.ModelsAndRenderers">MyChartRenderer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Displays bar chart.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Pair.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Pair</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the aligned site id's of the reference and query cmaps.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/ServerHandling/ProgressMonitor.html" title="class in ComparativeGenomics.ServerHandling">ProgressMonitor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A custom <code>JProgressBar</code> that will be populated when a file being
 uploaded to an external server has completed.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/QryContig.html" title="class in DataTypes">QryContig</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Query.html" title="class in DataTypes">Query</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Panels/QueryPanel.html" title="class in ComparativeGenomics.Panels">QueryPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Displays the alignments of a query for a chromosome.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Drawing/QueryShape.html" title="class in ComparativeGenomics.Drawing">QueryShape</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Draws the query chromosome that has been aligned to the reference chromosome
 in <code>QueryPanel</code></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="MapOptics/QueryView.html" title="class in MapOptics">QueryView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/Default/QueryViewData.html" title="class in Datasets.Default">QueryViewData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/Default/RawFileData.html" title="class in Datasets.Default">RawFileData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/RefContig.html" title="class in DataTypes">RefContig</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Reference.html" title="class in DataTypes">Reference</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="MapOptics/ReferenceView.html" title="class in MapOptics">ReferenceView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/Default/RefViewData.html" title="class in Datasets.Default">RefViewData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="startScreen/runMapOptics.html" title="class in startScreen">runMapOptics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains main function to run MapOptics.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/UserEdited/SavedQryData.html" title="class in Datasets.UserEdited">SavedQryData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/UserEdited/SavedRefData.html" title="class in Datasets.UserEdited">SavedRefData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/UserEdited/SearchRegionData.html" title="class in Datasets.UserEdited">SearchRegionData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Sequence.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Sequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains a nucleotide sequence and calculates useful statistics such as
 number of Ns and GC content</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/Site.html" title="class in ComparativeGenomics.FileHandling.DataHandling">Site</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Store the information of a digestion site on a cmap contig
 Any match event associated with this site is stored in a HashMap
 The key is the xmap id, the value is an ArrayList of match objects</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Smap.html" title="class in ComparativeGenomics.FileHandling">Smap</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Read and parse SV detection outputs from FaNDOM or RefAligner.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Algorithms/SortOverlap.html" title="class in Algorithms">SortOverlap</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Algorithms/SortSequence.html" title="class in Algorithms">SortSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/ServerHandling/SSH.html" title="class in ComparativeGenomics.ServerHandling">SSH</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Enables SSH connection to an external server, commands execution, creation of Docker container,
 running of scripts run_job.sh and calc_best_enz.sh as well as file uploading and downloading using STFP.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="startScreen/startScreen.html" title="class in startScreen">startScreen</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">First screen displayed when opening MapOptics.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="MapOptics/SummaryView.html" title="class in MapOptics">SummaryView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/Default/SummaryViewData.html" title="class in Datasets.Default">SummaryViewData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/SV.html" title="class in DataTypes">SV</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/SVFandom.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVFandom</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Saves all the information contained in the file SV.txt, output
 of FaNDOM alignments in which SV information are saved.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/SVRefAligner.html" title="class in ComparativeGenomics.FileHandling.DataHandling">SVRefAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Saves all the information contained in the file SMAP, output
 from runBNG SV, that contains data about SVs detected during the alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="MapOptics/SVView.html" title="class in MapOptics">SVView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/Default/SVViewData.html" title="class in Datasets.Default">SVViewData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="UserInterface/ModelsAndRenderers/TableModels.html" title="class in UserInterface.ModelsAndRenderers">TableModels</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Models of tables</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/StructuralVariant/Translocation.html" title="class in ComparativeGenomics.StructuralVariant">Translocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores translocations data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="DataTypes/Translocation.html" title="class in DataTypes">Translocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/UserEdited/UserQryData.html" title="class in Datasets.UserEdited">UserQryData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="Datasets/UserEdited/UserRefData.html" title="class in Datasets.UserEdited">UserRefData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="Algorithms/Variants.html" title="class in Algorithms">Variants</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/Xmap.html" title="class in ComparativeGenomics.FileHandling">Xmap</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Verifies file format, parse and save the contents of an XMAP file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/FileHandling/DataHandling/XmapData.html" title="class in ComparativeGenomics.FileHandling.DataHandling">XmapData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores the data for each individual alignment identified
 as the XMAP file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="FileHandling/XmapReader.html" title="class in FileHandling">XmapReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="ComparativeGenomics/Drawing/XmapShape.html" title="class in ComparativeGenomics.Drawing">XmapShape</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Draws the alignment to the panel like this: _________ |
 | | | | | |-------| |-------| Where the lines to join the alignment are
 anchored to the site positions on the reference and the query can be moved
 about, with the alignment lines following.</div>
</div>
</div>
</div>
</main>
</div>
</div>
</body>
</html>
